plutus-benchmark/bitwise/test/Main.hs

```diff
@@ -1,12 +1,72 @@
+-- editorconfig-checker-disable-file
+{-# LANGUAGE OverloadedStrings #-}
+
 module Main (main) where
 
+import Crypto.Hash.SHA512 qualified as SHA512Ref
+import Crypto.Sign.Ed25519 (PublicKey (PublicKey), Signature (Signature))
+import Crypto.Sign.Ed25519 qualified as Ed25519Ref
+import PlutusBenchmark.Ed25519 (checkValid)
+import PlutusBenchmark.Ed25519.Compiled (checkValidCompiled, msgAsData, pkAsData, signatureAsData)
 import PlutusBenchmark.NQueens (nqueens)
-import Test.Tasty (defaultMain, testGroup)
+import PlutusBenchmark.NQueens.Compiled (dimAsData, nqueensCompiled)
+import PlutusBenchmark.SHA512 (sha512)
+import PlutusTx.Builtins (fromBuiltin, toBuiltin)
+import PlutusTx.Code (unsafeApplyCode)
+import PlutusTx.Test (goldenBudget, goldenEvalCekCatch, goldenPirReadable, goldenSize)
+import Test.Tasty (TestTree, defaultMain, testGroup)
+import Test.Tasty.Extras (TestNested, runTestNested, testNestedGhc)
 import Test.Tasty.HUnit (assertEqual, testCase)
 
 main :: IO ()
-main = defaultMain . testGroup "nqueens" $ [
-  testCase "solves for 8 queens" $ assertEqual ""
-    [(0,0), (1,4), (2,7), (3,5), (4,2), (5,6), (6,1), (7,3)]
-    (nqueens 8)
+main = defaultMain . testGroup "bitwise" $ [
+  testGroup "N-queens" [
+    testCase "solves for 8 queens" $ assertEqual ""
+      [(0,0), (1,4), (2,7), (3,5), (4,2), (5,6), (6,1), (7,3)]
+      (nqueens 8),
+    runTestGhc [
+      goldenPirReadable "8 queens" $ nqueensCompiled `unsafeApplyCode` dimAsData,
+      goldenSize "8 queens" $ nqueensCompiled `unsafeApplyCode` dimAsData,
+      goldenBudget "8 queens" $ nqueensCompiled `unsafeApplyCode` dimAsData,
+      goldenEvalCekCatch "8 queens" [nqueensCompiled `unsafeApplyCode` dimAsData]
+      ]
+    ],
+  testGroup "Ed25519" [
+    testCase "SHA512 works" sha512Case,
+    testCase "Ed25519 works" ed25519Case,
+    runTestGhc [
+      goldenPirReadable "Ed25519" $
+        checkValidCompiled `unsafeApplyCode` signatureAsData `unsafeApplyCode` msgAsData `unsafeApplyCode` pkAsData,
+      goldenSize "Ed25519" $
+        checkValidCompiled `unsafeApplyCode` signatureAsData `unsafeApplyCode` msgAsData `unsafeApplyCode` pkAsData,
+      goldenBudget "Ed25519" $
+        checkValidCompiled `unsafeApplyCode` signatureAsData `unsafeApplyCode` msgAsData `unsafeApplyCode` pkAsData,
+      goldenEvalCekCatch "Ed25519" [
+        checkValidCompiled `unsafeApplyCode` signatureAsData `unsafeApplyCode` msgAsData `unsafeApplyCode` pkAsData
+        ]
+      ]
+    ]
   ]
+
+-- Cases
+
+sha512Case :: IO ()
+sha512Case = do
+  let testBS = "hello world"
```

Comment:
A few more test cases for SHA-512 here wouldn't do any harm given the complexity of the implementation!